catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -778,6 +832,20 @@ protected void addCatalogSourceDescriptor(Set<SourceDescriptor> descriptors) {
         }
     }
 
+    private void buildPolicyMap(HashMap<String, Set<String>> itemPolicyMap,
+            Set<Entry<String, Set<String>>> policy) {
+        if (policy != null) {
```

Comment:
If you could guarantee that policy and access would be empty lists instead of null (which would be more intuitive too) this code could look at lot more streamlined getting rid of the null checks.
```java
if (policy != null) {
```